### PR TITLE
Add Nexmo as a text provider for MX and US numbers

### LINKED ISSUE #926: Users SBAT receive text messages fast in Mexico via a Twilio alternative
### Expected Behavior

We should explore more local Twilio alternatives to assess if performance can be improved

### Current Behavior

Text messages sent via Twilio exhibit large variance in delivery times in Mexico

### PR DESCRIPTION
**Pull request by nambrot**
### Description

This PR adds the ability for us to send text messages via Nexmo for US and MX phone numbers.

### Tested

- Deployed it to integration and saw it working

### Related issues

- Fixes #926

### CODE REVIEW

**Review by cmcewen — APPROVED** [review]
LGTM but seems like some tests to fix

File: packages/verification-pool-api/src/config.ts [hunk 1]
```diff
@@ -35,6 +38,38 @@ export function getTwilioClient() {
...
+  if (nexmoClient == null) {
+    nexmoClient = new Nexmo({
+      apiKey: functionConfig.shared['nexmo-key'],
+      apiSecret: functionConfig.shared['nexmo-secret'],
+    })
+  }
+  return nexmoClient
+}
+
+export async function sendSmsWithNexmo(countryCode: string, phoneNumber: string, message: string) {
```
Comment (jmrossy) [inline packages/verification-pool-api/src/config.ts:51]: Nit: Config.ts seems like the wrong place for this

File: packages/verification-pool-api/src/verification.ts [hunk 2]
```diff
@@ -70,18 +71,31 @@ function getFormattedMessage(message: string) {
...
+      console.info('Message sent via Twilio')
+    } else {
+      await sendSmsWithNexmo(countryCode, phoneNumber, messageText)
+      console.info('Message sent via Nexmo')
+    }
   } catch (e) {
-    console.error('Failed to send twilio message', e)
-    throw new Error('Failed to send twilio message' + e)
+    console.error('Failed to send text message via txt provider', e)
+    throw new Error('Failed to send text message via txt provider' + e)
```
Comment (jmrossy) [inline packages/verification-pool-api/src/verification.ts:98]: Nit: this is probably my mistake but we shouldn't include `e` in the new error here

File: packages/verification-pool-api/src/config.ts (hunk 1 above)
Comment (nambrot) [reply packages/verification-pool-api/src/config.ts:51]: Given the relative urgency of this, do you think we can do this in a future PR? I imagine the pool will receive significant love (i.e. rewrite soon)

File: packages/verification-pool-api/src/verification.ts (hunk 2 above)
Comment (nambrot) [reply packages/verification-pool-api/src/verification.ts:98]: I think including more error information than not is usually more helpful, so I personally actually like it

File: packages/verification-pool-api/src/config.ts (hunk 1 above)
Comment (jmrossy) [reply packages/verification-pool-api/src/config.ts:51]: ok

File: packages/verification-pool-api/src/verification.ts (hunk 2 above)
Comment (jmrossy) [reply packages/verification-pool-api/src/verification.ts:98]: I feel like it just adds noise given that we properly log the error in the line above but it's really nbd

**Review by jmrossy — APPROVED** [review]

**Pull request merged**